shell.nix

```diff
@@ -1,19 +1,43 @@
 scope@{ pkgs ? import <nixpkgs> { } }:
+with pkgs;
+let
+  tokio-console = import ./default.nix { inherit pkgs; };
 
-let env = (import ./default.nix scope);
+  haveGlibcLocales = pkgs.glibcLocales != null && stdenv.hostPlatform.libc == "glibc";
+  glibcLocales-utf8 = (glibcLocales.override { locales = [ "en_US.UTF-8" ]; });
 
-in with pkgs;
+  env = buildEnv {
+    name = "console-env";
+    paths = [
+      direnv
+      binutils
+      stdenv
+      bashInteractive
+      docker
+      cacert
+      gcc
+      cmake
+      pkg-config
+      openssl
```

Comment:
Not actually sure if any of this is needed. 

* direnv should already exist, since it seems a little silly to start a nix shell manually, load the direnv hook from the shell environment, and then let direnv take over from there.
* Most of the build tools seem to either be unnecessary or already brought in by the tokio-console inputs.

Docker seems like the only useful thing that this adds.

Reply:
we probably don't need docker, fwiw, it was only there because i copied and pasted from a different nix derivation, and forgot to remove it. we should probably remove the docker dep!

Reply:
Cool, I'll clean this up then :+1: